### PR TITLE
Handle the ContractReceiveUpdateTransfer state change

### LINKED ISSUE #2109: Handle the ContractReceiveUpdateTransfer state change
Is not being dispatched and handled at the moment ContractReceiveUpdateTransfer

### PR DESCRIPTION
**Pull request by palango**
Fixes #2109

### CODE REVIEW

File: raiden/blockchain_events_handler.py [hunk 1]
```diff
@@ -215,6 +216,27 @@ def handle_channel_closed(raiden, event, current_block_number):
...
+
+    channel_state = views.get_channelstate_by_token_network_identifier(
+        views.state_from_raiden(raiden),
+        token_network_identifier,
+        channel_identifier,
+    )
+
+    if channel_state:
+        channel_transfer_updated = ContractReceiveUpdateTransfer(
+            transaction_from=data['closing_participant'],
```
Comment (hackaugusto) [inline raiden/blockchain_events_handler.py]: what about the address of the transaction sender?

Comment (palango) [reply raiden/blockchain_events_handler.py]: It's not at the event. As agreed in the office we should store the transaction hash in the future.

File: raiden/transfer/channel.py [hunk 2]
```diff
@@ -1722,11 +1724,32 @@ def handle_channel_closed(
...
 
+def handle_channel_updated_transfer(
+        channel_state: NettingChannelState,
+        state_change: ContractReceiveUpdateTransfer,
+        block_number: typing.BlockNumber,
+) -> TransitionResult:
+    if state_change.channel_identifier == channel_state.identifier:
+        # update transfer was called, make sure we don't call it again
+        channel_state.update_transaction = TransactionExecutionStatus(
+            started_block_number=block_number,
```
Comment (hackaugusto) [inline raiden/transfer/channel.py]: this is wrong, the block in which the transaction start is not available, only the sender knows it

File: raiden/transfer/channel.py [hunk 3]
```diff
@@ -1722,11 +1724,32 @@ def handle_channel_closed(
...
+            finished_block_number=block_number,
```
Comment (hackaugusto) [inline raiden/transfer/channel.py]: humm, this should probably be removed, we are not using the event block number in the state change, but the chainstate's block number, which may be significantly behind during a restart.

but this is for another issue/PR

File: raiden/transfer/channel.py (hunk 2 above)
Comment (palango) [reply raiden/transfer/channel.py]: Done

File: raiden/tests/unit/test_channelstate.py [hunk 4]
```diff
@@ -1579,3 +1580,67 @@ def test_update_must_be_called_if_close_lost_race():
...
+        channel_state.identifier,
+        23,
+    )
+
+    # update_transaction in channel state should not be set
+    channel_state = iteration.new_state
+    assert channel_state.update_transaction is None
+
+    update_block_number = 13
+    iteration2 = channel.handle_channel_updated_transfer(
```
Comment (hackaugusto) [inline raiden/tests/unit/test_channelstate.py]: this is weird, the channel is opened in the test, and update transfer may only be called [on a closed channel](https://github.com/raiden-network/raiden-contracts/blob/a2f9abbf0d0964b3e73fbd24dc09bcfbfaa94b1a/raiden_contracts/contracts/TokenNetwork.sol#L508).

Please fix the order of the state changes.

**Pull request merged**